graphql_compiler/compiler/ir_lowering_match.py

```diff
@@ -378,6 +436,114 @@ def remove_backtrack_blocks_from_fold(folded_ir_blocks):
     return new_folded_ir_blocks
 
 
+def _expression_list_to_conjunction(expression_list):
+    """Return an Expression that is the `&&` of all the expressions in the given list."""
+    if not isinstance(expression_list, list):
+        raise AssertionError(u'Expected list. Received {}: '
+                             u'{}'.format(type(expression_list).__name__, expression_list))
+    if len(expression_list) == 0:
+        raise AssertionError(u'Received empty expression_list '
+                             u'(function should never be called with empty list): '
+                             u'{}'.format(expression_list))
+    elif len(expression_list) == 1:
+        return expression_list[0]
+    else:
+        remaining_conjunction = _expression_list_to_conjunction(expression_list[1:])
+        return BinaryComposition(u'&&', expression_list[0], remaining_conjunction)
+
+
+def _extract_conjuction_elements_from_expression(expression):
+    """Return a list of expressions that are connected by `&&`s in the given expression."""
+    if isinstance(expression, BinaryComposition) and expression.operator == u'&&':
+        for element in _extract_conjuction_elements_from_expression(expression.left):
+            yield element
+        for element in _extract_conjuction_elements_from_expression(expression.right):
+            yield element
+    else:
+        yield expression
+
+
+def _construct_field_operator_expression_dict(expression_list, operators):
+    """Construct a mapping from local fields to specified operators, and corresponding expressions.
+
+    Args:
+        expression_list: list of expressions to analyze
+        operators: list of BInaryComposition operators used to select the "important" expressions
+
+    Returns:
+        local_field_to_expressions:
+            dict mapping local field names to "operator => BinaryComposition" dictionaries,
+            for each BinaryComposition operator involving the LocalField
+        remaining_expression_list:
+            list of remaining expressions that were *not*
+            BinaryCompositions on a LocalField using any of the specified operators
+    """
+    local_field_to_expressions = {}
+    remaining_expression_list = deque([])
+    for expression in expression_list:
+        if all([isinstance(expression, BinaryComposition),
```

Comment:
Prefer to use tuples rather than lists for `all()`:
`if all((option_a, option_b, option_c)):`

It's also a little strange that only `expression.left` is considered. Can we make this symmetrical? Otherwise, a lowering change elsewhere could make this pass stop working.